src/ixdat/readers/biologic.py

```diff
@@ -1,9 +1,212 @@
+        if self.place_in_file == "header":
+            self.process_header_line(line)
+        elif self.place_in_file == "column names":
+            self.process_column_line(line)
+        elif self.place_in_file == "data":
+            self.process_data_line(line)
+        else:  # just for debugging
+            raise ReadError(f"place_in_file = {self.place_in_file}")
+        self.n_line += 1
+
+    def process_header_line(self, line):
+        """Search line for important metadata and set the relevant attribute of self"""
+        self.header_lines.append(line)
+        if not self.N_header_lines:
+            N_head_match = re.search(regular_expressions["N_header_lines"], line)
+            if N_head_match:
+                self.N_header_lines = int(N_head_match.group(1))
+                return
+        if self.n_line == 3:
+            self.ec_technique = line.strip()
+            return
+        if not self.timestamp_string:
+            timestamp_match = re.search(regular_expressions["timestamp_string"], line)
+            if timestamp_match:
+                self.timestamp_string = timestamp_match.group(1)
+                self.tstamp = timestamp_string_to_tstamp(self.timestamp_string)
+            return
+        loop_match = re.search(regular_expressions["loop"], line)
+        if loop_match:
+            print(f"loop specified on line='{line}'")
+            n = int(loop_match.group(1))
+            start = int(loop_match.group(2))
+            finish = int(loop_match.group(3))
+            if "loop_number" not in self.column_data:
+                self.column_data["loop_number"] = np.array([])
+            self.column_data["loop_number"] = np.append(
+                self.column_data["loop_number"], np.array([n] * (finish - start + 1))
+            )
+            return
+
+        if self.N_header_lines and self.n_line >= self.N_header_lines - 2:
+            self.place_in_file = "column names"
+
+    def process_column_line(self, line):
+        """Split the line to get the names of the file's data columns"""
+        self.header_lines.append(line)
+        self.column_names = line.strip().split(delim)
+        self.column_data.update({name: np.array([]) for name in self.column_names})
+        self.place_in_file = "data"
+
+    def process_data_line(self, line):
+        """Split the line and append the numbers the corresponding data column arrays"""
+        data_strings_from_line = line.strip().split()
+        for name, value_string in zip(self.column_names, data_strings_from_line):
+            try:
+                value = float(value_string)
+            except ValueError:
+                if "," in value_string:  # oh my god, why?!
+                    value_string = value_string.replace(",", ".")
+                if "E" in value_string:  # Biologic uses capital E for sci. notation.
+                    value_string = value_string.replace(",", ".")
```

Comment:
The replace is the same as above. On purpose?

Reply:
Nope. Should have been replace "E" with "e". That it works with this error means that `float(float_as_str)` works fine if float_as_str has "E". Deleted the second if and it works fine.